.github/actions/free-hosted-runner-disk-space/action.yml

```diff
@@ -1,18 +1,148 @@
 name: "Free GHA runner disk space"
 description: "Removes unnecessary files from the runner."
 inputs:
+  only-remove-files:
+    description: 'Only remove files and folders, do not manipulate the runner memory'
+    required: false
+    default: 'true'
+  root-reserve-mb:
+    description: 'Space to be left free on the root filesystem, in Megabytes.'
+    required: false
+    default: '10240'
+  temp-reserve-mb:
+    description: 'Space to be left free on the temp filesystem (/mnt), in Megabytes.'
+    required: false
+    default: '100'
+  swap-size-mb:
+    description: 'Swap space to create, in Megabytes.'
+    required: false
+    default: '1024'
+  overprovision-lvm:
+    description: |
+      Create the LVM disk images as sparse files, making the space required for the LVM image files *appear* unused on the
+      hosting volumes until actually allocated. Use with care, this can lead to surprising out-of-disk-space situations.
+      You should prefer adjusting root-reserve-mb/temp-reserve-mb over using this option.
+    required: false
+    default: 'false'
+  build-mount-path:
+    description: 'Absolute path to the mount point where the build space will be available, defaults to /var/lib/docker/ if unset.'
+    required: false
+  build-mount-path-ownership:
+    description: 'Ownership of the mount point path, defaults to standard "runner" user and group.'
+    required: false
+    default: 'runner:runner'
+  pv-loop-path:
+    description: 'Absolute file path for the LVM image created on the root filesystem, the default is usually fine.'
+    required: false
+    default: '/pv.img'
+  tmp-pv-loop-path:
+    description: 'Absolute file path for the LVM image created on the temp filesystem, the default is usually fine. Must reside on /mnt'
+    required: false
+    default: '/mnt/tmp-pv.img'
+  remove-dotnet:
+    description: 'Removes .NET runtime and libraries. (frees ~17 GB)'
+    required: false
+    default: 'true'
+  remove-android:
+    description: 'Removes Android SDKs and Tools. (frees ~11 GB)'
+    required: false
+    default: 'false'
+  remove-haskell:
+    description: 'Removes GHC (Haskell) artifacts. (frees ~2.7 GB)'
+    required: false
+    default: 'true'
+  remove-codeql:
+    description: 'Removes CodeQL Action Bundles. (frees ~5.4 GB)'
+    required: false
+    default: 'false'
+  remove-boost:
+    description: 'Removes Boost C++ libraries. (frees ~1 GB)'
+    required: false
+    default: 'true'
+  remove-tools-cache:
+    description: 'Removes tools and dependencies cached. (frees ~3 GB)'
+    required: false
+    default: 'true'
+  remove-docker-images:
+    description: 'Removes cached Docker images. (frees ~3 GB)'
+    required: false
+    default: 'false'
   to-remove:
-    description: Collection of files/folders to be removed
-    default: "/usr/share/dotnet /opt/ghc /usr/local/share/boost $AGENT_TOOLSDIRECTORY"
+    description: 'Additional collection of files/folders to be removed'
+    default: ""
```

Comment:
I don't see this `to-remove` being used anywhere anymore.
I think it's useful to retain the ability to dynamically define more folders to delete, outside of the well-known ones.
IMO it should be used again, especially so that this won't be a breaking change upgrade!